agent/cache/cache.go

```diff
@@ -278,8 +278,8 @@ RETRY_GET:
 
 // entryKey returns the key for the entry in the cache. See the note
 // about the entry key format in the structure docs for Cache.
-func (c *Cache) entryKey(r *RequestInfo) string {
-	return fmt.Sprintf("%s/%s/%s", r.Datacenter, r.Token, r.Key)
+func (c *Cache) entryKey(t string, r *RequestInfo) string {
+	return fmt.Sprintf("%s/%s/%s/%s", t, r.Datacenter, r.Token, r.Key)
```

Comment:
This is the fix for another hypothetical bug I found along the way. It was well discussed with Mitchell on Slack and the test case below highlights why it might be needed one day but can be ignored for purposes of reviewing the `blockingQuery` fix.

Reply:
Total nitpick but should the type be the prefix here? We don't do any cache key searching or lookups or anything fancy but least specificity => most usually has benefits. Super hypothetical but _feels_ better.

Reply:
Hmm thy type _is_ the prefix here. It seemed to me that type is the most coarse grained partitioning we'd want (cache results from different operation types are entirely separate data sets whereas split by DC or token or key are finer grained.

So I'd agree with your high level point but either I'm not understanding why you think the cache type is _less_ specific than the other elements, or there is confusion caused by the diff rendering highlighting the last `/%s` when really the new `t` value is being added as the _first_ segment.

Reply:
Oh right, the git diff just threw me off. Hah! Ignore me.